tst/EnergyPlus/unit/ChillerExhaustAbsorption.unit.cc

```diff
@@ -637,29 +638,60 @@ TEST_F(EnergyPlusFixture, ExhAbsorption_calcHeater_Fix_Test)
 
     thisChillerHeater.HWPlantLoc.loopNum = 1;
     thisChillerHeater.HWPlantLoc.loopSideNum = DataPlant::LoopSideLocation::Demand;
-    state->dataPlnt->PlantLoop(1).FluidName = "WATER";
-    state->dataPlnt->PlantLoop(1).FluidIndex = 1;
-    state->dataPlnt->PlantLoop(1).LoopDemandCalcScheme = DataPlant::LoopDemandCalcScheme::SingleSetPoint;
-    state->dataPlnt->PlantLoop(1).LoopSide(DataPlant::LoopSideLocation::Demand).FlowLock = DataPlant::FlowLock::Locked;
-    state->dataLoopNodes->Node(3).Temp = 60.0;
-    state->dataLoopNodes->Node(3).MassFlowRate = 0.5;
-    state->dataLoopNodes->Node(4).TempSetPoint = 70.0;
-    state->dataLoopNodes->Node(7).Temp = 350.0;
-    state->dataLoopNodes->Node(7).MassFlowRate = 0.5;
+    auto &hwPlantLoop = state->dataPlnt->PlantLoop(1);
+    hwPlantLoop.FluidName = "WATER";
+    hwPlantLoop.FluidIndex = 1;
+    hwPlantLoop.LoopDemandCalcScheme = DataPlant::LoopDemandCalcScheme::SingleSetPoint;
+    hwPlantLoop.LoopSide(DataPlant::LoopSideLocation::Demand).FlowLock = DataPlant::FlowLock::Locked;
+
+    EXPECT_EQ(1, thisChillerHeater.ChillReturnNodeNum);
+    EXPECT_EQ("EXH CHILLER INLET NODE", state->dataLoopNodes->NodeID(1));
+    EXPECT_EQ(2, thisChillerHeater.ChillSupplyNodeNum);
+    EXPECT_EQ("EXH CHILLER OUTLET NODE", state->dataLoopNodes->NodeID(2));
+    EXPECT_EQ(3, thisChillerHeater.HeatReturnNodeNum);
+    EXPECT_EQ("EXH CHILLER HEATING INLET NODE", state->dataLoopNodes->NodeID(3));
+    EXPECT_EQ(4, thisChillerHeater.HeatSupplyNodeNum);
+    EXPECT_EQ("EXH CHILLER HEATING OUTLET NODE", state->dataLoopNodes->NodeID(4));
+    EXPECT_EQ(5, thisChillerHeater.CondReturnNodeNum);
+    EXPECT_EQ("EXH CHILLER CONDENSER INLET NODE", state->dataLoopNodes->NodeID(5));
+    EXPECT_EQ("CAPSTONE C65 COMBUSTION AIR INLET NODE", state->dataLoopNodes->NodeID(6));
+    EXPECT_EQ(7, thisChillerHeater.ExhaustAirInletNodeNum);
+    EXPECT_EQ("CAPSTONE C65 COMBUSTION AIR OUTLET NODE", state->dataLoopNodes->NodeID(7));
+
+    constexpr Real64 hwSupplySetpoint = 70.0;
+    constexpr Real64 hwReturnTemp = 60.0;
+    constexpr Real64 hwMassFlow = 0.5;
+    state->dataLoopNodes->Node(thisChillerHeater.HeatReturnNodeNum).Temp = hwReturnTemp;
+    state->dataLoopNodes->Node(thisChillerHeater.HeatReturnNodeNum).MassFlowRate = hwMassFlow;
+    state->dataLoopNodes->Node(thisChillerHeater.HeatSupplyNodeNum).TempSetPoint = hwSupplySetpoint;
+
+    // Minimum temperature leaving the Chiller absorber is 176.6 C (350 F)
+    constexpr Real64 exhaustInTemp = 350.0;
+    constexpr Real64 absLeavingTemp = 176.667;
+    constexpr Real64 exhaustInMassFlowRate = 0.5;
+    constexpr Real64 exhaustInHumRate = 0.005;
+    state->dataLoopNodes->Node(thisChillerHeater.ExhaustAirInletNodeNum).Temp = exhaustInTemp;
+    state->dataLoopNodes->Node(thisChillerHeater.ExhaustAirInletNodeNum).MassFlowRate = exhaustInMassFlowRate;
+    state->dataLoopNodes->Node(thisChillerHeater.ExhaustAirInletNodeNum).HumRat = exhaustInHumRate;
```

Comment:
Setting the HumRat to non zero, so we catch the fact that CpAir was always calculated at 0.